libs/template/file.go

```diff
@@ -0,0 +1,96 @@
+package template
+
+import (
+	"context"
+	"io"
+	"io/fs"
+	"os"
+	"path/filepath"
+
+	"github.com/databricks/cli/libs/filer"
+)
+
+// Interface for an in memory representation of a file
```

Comment:
Well, accurate because both implementations represent a file and metadata for both are stored in memory :P

Changed the header comment to something clearer and more accurate